resources/todomvc/big-dom-generator/src/tree-generator.js

```diff
@@ -3,6 +3,39 @@ import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_
 
 const random = new LCG(DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR);
 
+const fillSubtreeWeights = (node, expandableItemWeight, nonExpandableItemWeight) => {
+    if (node.type === "expandableItem")
+        node.subTreeWeight = node.children.reduce((acc, child) => acc + fillSubtreeWeights(child, expandableItemWeight, nonExpandableItemWeight), expandableItemWeight);
+    else
+        node.subTreeWeight = nonExpandableItemWeight;
+
+    return node.subTreeWeight;
+};
+
+const markDisplayNoneNodes = (node, expandableItemWeight, nonExpandableItemWeight) => {
```

Comment:
I see you added a comment in the next commit, sorry I missed it! Thanks!
 It could be expanded with more explanations about the logic, especially an example of the result.

Reply:
Done.